src/test/plugins/mvcc_delete_plugin_system_test.cpp

```diff
@@ -239,13 +239,23 @@ TEST_F(MvccDeletePluginSystemTest, CheckPlugin) {
 
   // (11) Prepare clean-up of chunk 3
   {
+    // Wait for the previous updates to finish
+    {
+      auto attempts_remaining = max_attempts;
+      while (_counter < INITIAL_CHUNK_COUNT * CHUNK_SIZE - 2) {
+        std::this_thread::sleep_for(100);
+      }
+      // Check that we have not given up
+      ASSERT_GT(attempts_remaining, -1);
+    }
+
```

Comment:
Is this part going to stay after we fixed the TSAN issue?

Reply:
Yes. To check whether an async process has done its job, we either need to add hooks (which I don't like in the lib code if their sole purpose is to facilitate testing) or wait until we have externally visible effects.